src/jit/emitxarch.cpp

```diff
@@ -214,7 +217,8 @@ bool TakesRexWPrefix(instruction ins, emitAttr attr)
     // cased here.
     //
     // Rex_jmp = jmp with rex prefix always requires rex.w prefix.
-    if (ins == INS_movsx || ins == INS_rex_jmp)
+    // vpermq requires w bit to set to 1
+    if (ins == INS_movsx || ins == INS_rex_jmp || ins == INS_vpermq)
     {
```

Comment:
This doesn't seem right to me. The VEX.W prefix, I believe, is generally the AVX encoding of the corresponding REX.W prefix. We do have encodings that use the VEX.W prefix, and it is set based on the opcode size. I think it may be that the vpermq encoding needs to have the correct operand size, in which case I believe that the W bit would be set appropriately.

---
In reply to: [109974462](https://github.com/dotnet/coreclr/pull/10662#discussion_r109974462) [](ancestors = 109974462)